magento/magento2
# Rollback MAGETWO-96403 as it was closed as "NOT A BUG"

# Description (*)
See https://github.com/magento/magento2/issues/24422
Cannot enable "Apply to shipping amount" when you Apply "Fixed amount discount for whole cart"
It should be possible according to documentation (https://docs.magento.com/m2/ce/user_guide/marketing/price-rules-cart-create.html):
``Fixed amount discount for whole cart:
Discounts the entire cart by subtracting a fixed amount from the cart total. For example: Enter 10 in Discount Amount to subtract $10 from the cart total.By default, the discount applies only to the cart subtotal. To apply the discount to the subtotal and shipping separately, see Apply to Shipping Amount.``

According to the notes There was a ticket, MAGETWO-96403, that was closed as "NOT A BUG", however, some parts of the code made it in to the code base


### Fixed Issues
1. magento/magento2#24422: "Apply discount to Shipping Amount" disabled for "Fixed amount discount for whole cart" action

### Manual testing scenarios
From the Adminhtml:
1. Go to admin panel -> Marketing -> Cart Price Rules -> Add new rule
2. In Actions tab and set Apply to "Fixed amount discount for whole cart"
3. Set a coupon code, and set the fixed amount to a value, say $10
4. Create a flat rate shipping rate equal to #3, above. In my example $10
5. Create a product LESS with a price LESS THAN #3, above. Example: $5

From the front end:
1. add product to cart
2. Add coupon code to cart or checkout page.
3. Total should now equal price of product, $5, NOT the price of shipping

### Questions or comments
There were several other test cases included with this commit that appeared to be cleanup. I have left those, but will test to see if any of them now fail. Any insight on these is appreciated.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#29619: Rollback MAGETWO-96403 as it was closed as "NOT A BUG"
